[wasip2] incoming http handler

rajatjindal opened the pull request:
This PR is first of many to support wasip2 in spin go sdk.

rajatjindal commented:
I'll update this branch to use wasm-tools-go from bytecodealliance org as that repo has been moved.

rajatjindal commented:
Hi @adamreese @michelleN 

I believe this PR is ready for review. 🎉 

thank you

adamreese commented:
Adding an extra line to the response output gives me this error
```
2024-09-20T18:07:05.502974Z ERROR spin_trigger_http::server: Error processing request: guest invocation failed

Caused by:
    0: error while executing at wasm backtrace:
           0: 0x42c98 - wit-component:shim!indirect-wasi:io/streams@0.2.0-[method]output-stream.write
           1: 0x17a4b - main!interface:{Write:func:{slice:basic:uint8}{basic:int,named:error}}.Write$invoke
           2: 0x32eaf - main!fmt.Fprintln
           3: 0x34163 - main!wasi:http/incoming-handler@0.2.0#handle
    1: write exceeded budget
```

rajatjindal commented:
> Adding an extra line to the response output gives me this error
> 
> ```
> 2024-09-20T18:07:05.502974Z ERROR spin_trigger_http::server: Error processing request: guest invocation failed
> 
> Caused by:
>     0: error while executing at wasm backtrace:
>            0: 0x42c98 - wit-component:shim!indirect-wasi:io/streams@0.2.0-[method]output-stream.write
>            1: 0x17a4b - main!interface:{Write:func:{slice:basic:uint8}{basic:int,named:error}}.Write$invoke
>            2: 0x32eaf - main!fmt.Fprintln
>            3: 0x34163 - main!wasi:http/incoming-handler@0.2.0#handle
>     1: write exceeded budget
> ```

Hi Adam, thanks for pointing this out. I have pushed a fix for it and it seems to be working now.

adamreese reviewed (commented):
@rajatjindal This is a great start and I'm excited to have something to iterate on. To keep it consistent with the other Spin repos I feel like we should stick to GitHub Actions rather than introducing Dagger at this time. Could we break out Dagger to a separate conversation from this PR?

rajatjindal commented:
> Could we break out Dagger to a separate conversation from this PR?

yes absolutely. I will push updated PR.

rajatjindal commented:
Hi @adamreese 

I have pushed the following changes:

- removed `dagger` for now
- update wasm-tools-go to latest version (and new url `go.bytecodealliance.org`
- added basic workflow using vanilla GitHub actions

Please let me know if there are other things we should address before merging this.
